cpp/tests/transform/nans_to_null_test.cpp

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cudf/types.hpp>
+#include <cudf/column/column.hpp>
+#include <cudf/column/column_view.hpp>
+#include <tests/utilities/base_fixture.hpp>
+#include <cudf/transform.hpp>
+#include <tests/utilities/type_lists.hpp>
+#include <tests/utilities/column_utilities.hpp>
+#include <tests/utilities/column_wrapper.hpp>
+
+
+template <typename T>
+struct NaNsToNullTest : public cudf::test::BaseFixture {
+
+    void run_test(cudf::column input, cudf::column expected) {
+
+        auto got_mask = cudf::experimental::nans_to_nulls(input);
+        cudf::column got (static_cast<cudf::column>(input));
+        got.set_null_mask(std::move(*(got_mask.first)));
+
+        EXPECT_EQ(expected.null_count(), got_mask.second);
+
+        cudf::test::expect_columns_equal(expected.view(), got.view());
+    }
+
+    cudf::column create_expected (std::vector<T> input, std::vector<bool> mask = {}) {
+        std::vector<T> expected(input);
+        std::vector<bool> expected_mask;
+
+        if (mask.size() > 0) {
+            std::transform(input.begin(), input.end(), mask.begin(), std::back_inserter(expected_mask), 
+                    [](T val, bool validity){return (std::isnan(val) or validity == false)? false : true;});
+        } else {
+            std::transform(input.begin(), input.end(), std::back_inserter(expected_mask), 
+                    [](T val){return (std::isnan(val))? false : true;});
+        }
+
+        return static_cast<cudf::column>(cudf::test::fixed_width_column_wrapper<T> (expected.begin(), expected.end(), expected_mask.begin()));
+
+    }
+};
+
+using test_types = ::testing::Types<float, double>;
+
+TYPED_TEST_CASE(NaNsToNullTest, test_types);
+
+TYPED_TEST(NaNsToNullTest, WithMask){
+    using T = TypeParam;
+
+    std::vector<T> input = {1, NAN, 3, NAN, 5, NAN};
+    std::vector<bool> mask = {1,   1, 1,   1, 0,   0};
+    auto input_column = cudf::test::fixed_width_column_wrapper<T> (input.begin(), input.end(), mask.begin());
+    this->run_test(static_cast<cudf::column>(input_column), this->create_expected(input, mask));
```

Comment:
```suggestion
    auto expected_column = this->create_expected(input, mask);
    this->run_test(input_column->view(), expected_column->view());
```
Same comment about casting `fixed_width_column_wrapper` to `column` possibly invoking a copy. Maybe `run_test` should take `column_view` instead.